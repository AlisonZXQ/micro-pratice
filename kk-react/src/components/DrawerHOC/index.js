import React, { useState, useEffect } from 'react';
import { Drawer } from 'antd';
import { connect } from 'dva';
import MyIcon from '@components/MyIcon';
import styles from './index.less';

export default function DrawerHOC() {
  return (DiffComponent) => {
    function Index(props) {
      const { drawerIssueId } = props;
      const [visible, setVisible] = useState(false);

      useEffect(() => {
        if (drawerIssueId) {
          setVisible(true);
        } else {
          setVisible(false);
        }
      }, [drawerIssueId]);

      // 阻止外层事件的触发关闭
      return (<div onClick={(e) => { e.stopPropagation() }}>
        <Drawer
          className={styles.drawer}
          placement="right"
          closable={false}
          visible={visible}
          maskClosable={false}
          mask={false}
          keyboard={false}
          width={665}
          bodyStyle={{ padding: '0px' }}
        >
          <div className={styles.collapse} onClick={() => props.dispatch({ type: 'receipt/saveDrawerIssueId', payload: '' })}>
            <MyIcon
              type='icon-zhankaijiantou'
              className={styles.icon}
            />
          </div>
          <DiffComponent {...props} />
        </Drawer>
      </div>);
    }

    const mapStateToProps = (state) => {
      return {
        drawerIssueId: state.receipt.drawerIssueId,
      };
    };

    return connect(mapStateToProps)(Index);
  };

}
