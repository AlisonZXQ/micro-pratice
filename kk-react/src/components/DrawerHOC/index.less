.drawer {
  .collapse {
    position      : absolute;
    left          : 0px;
    background    : @color-blue-6;
    width         : 1px;
    height        : 100%;
    z-index       : 9999;
    vertical-align: center;
    cursor: pointer;
    display: none;

    .icon {
      width          : 24px;
      height         : 24px;
      border         : 1px @color-black-4 solid;
      display        : flex;
      align-items    : center;
      justify-content: center;
      border-radius  : 100%;
      background     : @color-white-1;
      cursor         : pointer;
      position       : absolute;
      top            : calc(50vh - 12px);
      left           : -12px;
    }
  }

  &:hover {
    .collapse {
      display: block;
    }
  }
}
