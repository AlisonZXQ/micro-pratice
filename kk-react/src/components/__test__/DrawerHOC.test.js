import React from 'react';
import { mount } from 'enzyme';
import mountTest from '@tests/shared/mountTest';
import DrawerHOC from '../DrawerHOC';

describe('DrawerHOC', () => {
  mountTest(DrawerHOC);

  const setup = () => {
    const mountWrapper = mount(<DrawerHOC />);

    return {
      mountWrapper,
    }
  }

  test('DrawerHOC should contain Drawer', () => {
    const { mountWrapper } = setup();
    expect(mountWrapper).toMatchSnapshot();
  })
})
