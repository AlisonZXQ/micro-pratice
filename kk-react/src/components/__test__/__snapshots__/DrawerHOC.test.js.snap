// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerHOC DrawerHOC should contain Drawer 1`] = `<DrawerHOC />`;
